sphinxcontrib/datatemplates/directive.py

```diff
@@ -146,7 +146,8 @@ def run(self):
 
         relative_resolved_path, absolute_resolved_path = env.relfn2path(source)
 
-        env.note_dependency(absolute_resolved_path)
+        if len(source) > 0:
```

Comment:
Since source is a string, this can just be "if source:"

Should we be trying to compute the resolved paths at all if source is empty?

Reply:
Yes, the resolved path is added to the loader_options. I think that's used for relative file reverence, maybe?

Reply:
Ah, yeah, good point.